src/bin/cql-stress-cassandra-stress/operation/mod.rs

```diff
@@ -21,3 +23,41 @@ fn recompute_seed(seed: i64, partition_key: &CqlValue) -> i64 {
         _ => todo!("Implement recompute_seed for other CqlValues"),
     }
 }
+
+fn validate_row(generated_row: &[CqlValue], query_result: QueryResult) -> Result<()> {
+    let rows = match query_result.rows {
+        Some(rows) => rows,
+        None => anyhow::bail!("Query result doesn't contain any rows.",),
+    };
+
+    let first_row = match rows.split_first() {
+        Some((first_row, remaining_rows)) => {
+            // Note that row-generation logic behaves in a way that given partition_key,
+            // there is exactly one row with this partition_key.
+            anyhow::ensure!(remaining_rows.is_empty(), "Multiple rows matched the key.",);
+            first_row
+        }
+        None => anyhow::bail!("Query result doesn't contain any rows.",),
+    };
+
+    anyhow::ensure!(
+        first_row.columns.len() == generated_row.len(),
+        "Generated row's length: {}, result row's length: {}",
+        generated_row.len(),
+        first_row.columns.len()
+    );
+
+    let result =
+        first_row
+            .columns
+            .iter()
+            .zip(generated_row.iter())
+            .all(|(maybe_result, expected)| match maybe_result {
+                Some(result) => result == expected,
+                // TODO: For now, we don't permit NULLs.
+                None => false,
+            });
+
+    anyhow::ensure!(result, "The data doesn't match.");
```

Comment:
How about printing those rows so that the user can see how the mismatch looks like?